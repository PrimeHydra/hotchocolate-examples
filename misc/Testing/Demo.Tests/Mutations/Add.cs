namespace Demo.Tests.Mutations;

public class Add : QueryTestBase
{
    [Fact]
    public async Task AddSparkyAndReadHimBack()
    {
        await ExecuteQueryAsync(
            @"
mutation {
  addPuppy(input: { name: ""Sparky"", breed: ""Beagle"" }) {
    id
  }
}
                ");

        var result = await ExecuteQueryAsync(
            @"
{
  puppy(id: ""Id_1"") {
    name
    breed
  }
}

                   
                ");

        result.MatchSnapshot();
    }

    [Fact]
    public async Task AddDaisyAndReadHerBack()
    {
        await ExecuteQueryAsync(
            @"
mutation {
  addPuppy(input: { name: ""Daisy"", breed: ""Golden Doodle"" }) {
    id
  }
}
                ");

        var result = await ExecuteQueryAsync(
            @"
{
  puppy(id: ""Id_1"") {
    name
    breed
  }
}

                   
                ");

        result.MatchSnapshot();
    }

}