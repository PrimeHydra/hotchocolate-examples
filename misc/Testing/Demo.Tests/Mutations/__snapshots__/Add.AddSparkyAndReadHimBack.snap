{
  "data": {
    "puppy": {
      "name": "Sparky",
      "breed": "Beagle"
    }
  }
}
